src/component/Calendar/CalendarContainer.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import Calendar from 'react-calendar';
+import styled from 'styled-components';
+import './custom-calendar.css'; // 커스텀 CSS 파일
+import { getTodoProgressForDate } from '../../utils/LocalStorageUtil';
+import CustomHeader from './components/CustomHeader';
+
+
+function CalendarContainer({ onDateChange }) {
+  const [date, setDate] = useState(new Date());
+  const [month, setMonth] = useState(date.getMonth());
+
+  const handleChange = (selectedDate) => {
+    setDate(selectedDate);
+    onDateChange(selectedDate); // 부모 컴포넌트로 선택된 날짜 전달
+  };
+  // 월 변경 핸들러
+  const handleMonthChange = (e) => {
+    const selectedMonth = parseInt(e.target.value);
+    const newDate = new Date(date.getFullYear(), selectedMonth, 1);
+    setDate(newDate);
+    setMonth(selectedMonth);
+  };
+  // 각 타일에 내용을 추가하는 함수
+  const tileContent = ({ date, view }) => {
+    if (view === 'month') { // 월간 뷰일 때만 표시
+      const { completed, total } = getTodoProgressForDate(date); // 성취도 계산
+      if (total > 0) { // 할 일이 있는 경우에만 표시
+        return (
+          <TodoProgress>
+            {completed}/{total}
+          </TodoProgress>
+        );
+      }
+    }
+    return null;
+  };
+
+  // 특정 조건에 따라 날짜에 클래스 이름을 할당하는 함수
+  const getTileClassName = ({ date, view }) => {
+    if (view === 'month') {
+      const { completed, total } = getTodoProgressForDate(date); // 성취도 계산
+      if (date === new Date().toISOString().split('T')[0]) {
+        return 'today'; // 오늘 날짜를 하이라이트
+        } else if (completed === total && completed > 0 && total > 0) {
+          return 'great'; // 모든 할 일이 완료된 경우
```

Comment:
지금 조건으로도 모든 할 일이 완료된 경우에 대한 if문이 충분히 잘 기능하지만 조건을 조금 더 간결하게 `completed === total && total > 0`로도 작성할 수 있을 것 같아요!

```suggestion
} else if (completed === total && total > 0) {
          return 'great'; // 모든 할 일이 완료된 경우
```